### PR TITLE
flash attn pip install

### LINKED ISSUE #414: flash-attn wheels
### ⚠️ Please check that this feature request hasn't been suggested before.

- [X] I searched previous [Ideas in Discussions](https://github.com/OpenAccess-AI-Collective/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [X] I searched previous [Issues](https://github.com/OpenAccess-AI-Collective/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

Now that flash-attn provides wheels, we can remove it from the docker base image and simply install it with requirements.txt

### ✔️ Solution

see above

### ❓ Alternatives

_No response_

### 📝 Additional Context

_No response_

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this feature has not been requested yet.
- [X] I have provided enough information for the maintainers to understand and evaluate this request.

### PR DESCRIPTION
**Pull request by mhenrichsen**
Untested.

Should pip install flash attn 2.0.8 instead of building wheels.

Fixes: https://github.com/OpenAccess-AI-Collective/axolotl/issues/414

### CODE REVIEW

**Comment by mhenrichsen** [comment]
I can't figure out why it can't find `packaging`...

File: docker/Dockerfile-base [hunk 1]
```diff
@@ -90,13 +72,8 @@ RUN mkdir -p /workspace/wheels/bitsandbytes
...
 COPY --from=bnb-builder /workspace/bitsandbytes/dist/bitsandbytes-*.whl wheels
 COPY --from=bnb-builder /workspace/bitsandbytes/bitsandbytes/libbitsandbytes*.so wheels/bitsandbytes
-COPY --from=flash-attn-builder /workspace/flash-attention/dist/flash_attn-*.whl wheels
-COPY --from=flash-attn-builder /workspace/flash-attention/csrc/fused_dense_lib/dist/fused_dense_lib-*.whl wheels
-COPY --from=flash-attn-builder /workspace/flash-attention/csrc/xentropy/dist/xentropy_cuda_lib-*.whl wheels
-COPY --from=flash-attn-builder /workspace/flash-attention/csrc/rotary/dist/rotary_emb-*.whl wheels
-COPY --from=flash-attn-builder /workspace/flash-attention/csrc/layer_norm/dist/dropout_layer_norm-*.whl wheels
 
-RUN pip3 install wheels/deepspeed-*.whl wheels/flash_attn-*.whl wheels/fused_dense_lib-*.whl wheels/xentropy_cuda_lib-*.whl wheels/rotary_emb-*.whl wheels/dropout_layer_norm-*.whl
+RUN pip3 install wheels/deepspeed-*.whl wheels/fused_dense_lib-*.whl wheels/xentropy_cuda_lib-*.whl wheels/rotary_emb-*.whl wheels/dropout_layer_norm-*.whl
```
Comment (winglian) [inline docker/Dockerfile-base]: Don't forget to remove the other wheels that flash attn builds.

**Review by winglian — CHANGES_REQUESTED** [review]
Also need to add flash-attn to the requirements.txt file. Thanks!

File: docker/Dockerfile-base (hunk 1 above)
Comment (mhenrichsen) [reply docker/Dockerfile-base]: These?
`wheels/fused_dense_lib-*.whl wheels/xentropy_cuda_lib-*.whl wheels/rotary_emb-*.whl wheels/dropout_layer_norm-*.whl`

**Comment by mhenrichsen** [comment]
> Also need to add flash-attn to the requirements.txt file. Thanks!

That's what I did to begin with, but it fails because it can't find `packaging`.

File: docker/Dockerfile-base (hunk 1 above)
Comment (mhenrichsen) [reply docker/Dockerfile-base]: Should be good now.

**Pull request merged**